"Handler <function mangle_docstrings at 0x7f64b5ba57b8> for event 'autodoc-process-docstring' threw an exception"

We're starting to see this with the upgrade from 0.9.2 to 1.0

For example: https://github.com/yeatmanlab/pyAFQ/pull/270/checks?check_run_id=800934528#step:6:61

See also: https://github.com/yeatmanlab/pyAFQ/pull/271

Do we need to change something on our end?

With `SPHINXOPTS="-nWT --keep-going -vvv"`, the traceback is: 


```
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/sphinx/events.py", line 110, in emit
    results.append(listener.handler(self.app, *args))
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/numpydoc/numpydoc.py", line 155, in mangle_docstrings
    'xref_aliases': app.config.numpydoc_xref_aliases_complete,
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/sphinx/config.py", line 250, in __getattr__
    raise AttributeError(__('No such config value: %s') % name)
AttributeError: No such config value: numpydoc_xref_aliases_complete

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/sphinx/cmd/build.py", line 279, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/sphinx/application.py", line 277, in __init__
    self._init_builder()
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/sphinx/application.py", line 333, in _init_builder
    self.events.emit('builder-inited')
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/sphinx/events.py", line 110, in emit
    results.append(listener.handler(self.app, *args))
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/autoapi/extension.py", line 129, in run_autoapi
    sphinx_mapper_obj.map(options=app.config.autoapi_options)
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/autoapi/mappers/python/mapper.py", line 275, in map
    super(PythonSphinxMapper, self).map(options)
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/autoapi/mappers/base.py", line 276, in map
    for obj in self.create_class(data, options=options):
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/autoapi/mappers/python/mapper.py", line 317, in create_class
    lines,
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/sphinx/application.py", line 450, in emit
    return self.events.emit(event, *args, allowed_exceptions=allowed_exceptions)
  File "/Users/arokem/.virtualenvs/afq/lib/python3.7/site-packages/sphinx/events.py", line 118, in emit
    (listener.handler, name)) from exc
sphinx.errors.ExtensionError: Handler <function mangle_docstrings at 0x11246fe18> for event 'autodoc-process-docstring' threw an exception

Extension error:
Handler <function mangle_docstrings at 0x11246fe18> for event 'autodoc-process-docstring' threw an exception
make: *** [html] Error 2
```

Indeed, if I add a call to `update_config(app)` [here](https://github.com/numpy/numpydoc/blob/master/numpydoc/numpydoc.py#L145) this error is not raised.

--- a/numpydoc/numpydoc.py
+++ b/numpydoc/numpydoc.py
@@ -218,7 +218,7 @@ def setup(app, get_doc_object_=get_doc_object):
 
     app.setup_extension('sphinx.ext.autosummary')
 
-    app.connect('builder-inited', update_config)
+    app.connect('config-inited', update_config)
     app.connect('autodoc-process-docstring', mangle_docstrings)
     app.connect('autodoc-process-signature', mangle_signature)
     app.connect('doctree-read', relabel_references)
@@ -244,17 +244,19 @@ def setup(app, get_doc_object_=get_doc_object):
     return metadata
 
 
-def update_config(app):
+def update_config(app, config=None):
     """Update the configuration with default values."""
+    if config is None:  # only really needed for testing
+        config = app.config
     # Do not simply overwrite the `app.config.numpydoc_xref_aliases`
     # otherwise the next sphinx-build will compare the incoming values (without
     # our additions) to the old values (with our additions) and trigger
     # a full rebuild!
-    numpydoc_xref_aliases_complete = deepcopy(app.config.numpydoc_xref_aliases)
+    numpydoc_xref_aliases_complete = deepcopy(config.numpydoc_xref_aliases)
     for key, value in DEFAULT_LINKS.items():
         if key not in numpydoc_xref_aliases_complete:
             numpydoc_xref_aliases_complete[key] = value
-    app.config.numpydoc_xref_aliases_complete = numpydoc_xref_aliases_complete
+    config.numpydoc_xref_aliases_complete = numpydoc_xref_aliases_complete
 
 
 # ------------------------------------------------------------------------------

--- a/numpydoc/numpydoc.py
+++ b/numpydoc/numpydoc.py
@@ -28,6 +28,7 @@ from docutils.nodes import citation, Text, section, comment, reference
 import sphinx
 from sphinx.addnodes import pending_xref, desc_content
 from sphinx.util import logging
+from sphinx.errors import ExtensionError
 
 if sphinx.__version__ < '1.6.5':
     raise RuntimeError("Sphinx 1.6.5 or newer is required")
@@ -218,7 +219,13 @@ def setup(app, get_doc_object_=get_doc_object):
 
     app.setup_extension('sphinx.ext.autosummary')
 
-    app.connect('config-inited', update_config)
+    # Once we bump our Sphinx requirement higher (1.7 or 1.8?)
+    # we can just connect to config-inited
+    try:
+        app.connect('config-inited', update_config)
+    except ExtensionError:
+        app.connect('builder-inited', update_config)
+
     app.connect('autodoc-process-docstring', mangle_docstrings)
     app.connect('autodoc-process-signature', mangle_signature)
     app.connect('doctree-read', relabel_references)
@@ -246,7 +253,7 @@ def setup(app, get_doc_object_=get_doc_object):
 
 def update_config(app, config=None):
     """Update the configuration with default values."""
-    if config is None:  # only really needed for testing
+    if config is None:  # needed for testing and old Sphinx
         config = app.config
     # Do not simply overwrite the `app.config.numpydoc_xref_aliases`
     # otherwise the next sphinx-build will compare the incoming values (without

